### PR TITLE
NgxTab

### PR DESCRIPTION
**Pull request by thanhtranit**

### CODE REVIEW

File: components/tab/tab-view.component.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    })
+    .Class(new _ngxTabView());
+
+    function _ngxTabView() {
+      this.constructor = function() {
+        this._active = false;
+        this.onShow = new ng.core.EventEmitter();
+        this.onHide = new ng.core.EventEmitter();
+
+        Object.defineProperty(this.__proto__, "active", {
```
Comment (Minh-Van) [inline components/tab/tab-view.component.js]: `this.__proto__` is deprecated. Use `Object.getPrototypeOf(this)` for replacment

File: components/tab/tab.component.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+          panes.toArray().map(function(p){
+            if(isNaN(p.index)){
+              p.index = ++max;
+            }
+          });
+        });
+
+        this.panes = panes;
+
+        Object.defineProperty(this.__proto__, "active", {
```
Comment (Minh-Van) [inline components/tab/tab.component.js]: `this.__proto__` is deprecated. Use `Object.getPrototypeOf(this)` for replacment

File: components/tab/tests/ui/isolated-components/test-case.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+    this.addDetail = function() {
+      this.id++;
+      this.details.push({
+        title: `Detail ${this.id}`,
+        text: `Some detail text for ${this.id}...`
+      });
+    };
+
+    this.removeDetail = function(detail){
```
Comment (Minh-Van) [inline components/tab/tests/ui/isolated-components/test-case.js]: After removing a tab, you should make ngx-tab select the first tab automatically

File: components/tab/tests/ui/isolated-components/test-case.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+      setTimeout(function(){
+        console.log('change index to 0');
+        _this.index = 0;
+      }, 1000);
+    };
+
+    this.addDetail = function() {
+      this.id++;
+      this.details.push({
+        title: `Detail ${this.id}`,
```
Comment (Minh-Van) [inline components/tab/tests/ui/isolated-components/test-case.js]: I don't think IE can understand template literal syntax

File: components/tab/tab.component.js (hunk 2 above)
Comment (huuthonguyen76) [inline components/tab/tab.component.js]: `panes.toArray().map(function(p){` ---> What is `p`?
This name is meanless. Can we use `pane`?

File: components/tab/tests/ui/isolated-components/test-case.js (hunk 4 above)
Comment (huuthonguyen76) [inline components/tab/tests/ui/isolated-components/test-case.js]: `var _this = this;` and then use `this`?

**Pull request closed without merging**